lib/http2Request.js

```diff
@@ -0,0 +1,45 @@
+/*!
+ * express
+ * Copyright(c) 2009-2013 TJ Holowaychuk
+ * Copyright(c) 2013 Roman Shtylman
+ * Copyright(c) 2014-2015 Douglas Christopher Wilson
+ * MIT Licensed
+ */
+
+'use strict';
+
+/**
+ * Module dependencies.
+ * @private
+ */
+var http2 = require('http2');
+var http2Constants = http2.constants;
+var decorator = require('./requestDecorator');
+var defineGetter = require('./utils').defineGetter;
+var http2Req = decorator(Object.create(http2.Http2ServerRequest.prototype));
+var parse = require('parseurl');
+/**
+ * Override default http2 path to return only pathname
+ * as it returns full url
+ * Gets the full URL from the headers
+ * Creates a mock object for parse to cache on and read from
+ * as url in http2 refers to this.path which will results in an error

```

Comment:
@jasnell this looks fine to me for compatibility, pinging you in to take a look :)

Reply:
As noted below, I don't think this is something we want to do; one of the key aspects od Express (and this is so key that all official middleware are written this way) is that you can write anything using onlt the base type and you can plop that in Express and it'll work 100% the same. If you look like pretty much all our middleware, we don't even test them against Express because of this core philosophy.